### PR TITLE
feat(MRS): add a new datasource to get clusters

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
Running tool: /usr/local/go/bin/go test -timeout 9000000000s -run ^TestAccDatasourceMrsClusters_basic$ github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/mrs -v

=== RUN   TestAccDatasourceMrsClusters_basic
=== PAUSE TestAccDatasourceMrsClusters_basic
=== CONT  TestAccDatasourceMrsClusters_basic
--- PASS: TestAccDatasourceMrsClusters_basic (695.83s)
PASS
ok  	github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/mrs	695.880s

```

### CODE REVIEW

File: docs/data-sources/mapreduce_clusters.md [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+
+* `external_ip` - External IP address.
+
+* `external_alternate_ip` - Backup external IP address.
+
+* `internal_ip` - Internal IP address.
+
+* `deployment_id` - Cluster deployment ID.
+
+* `remark` - Cluster remarks.
```
Comment (ShiChangkuo) [inline docs/data-sources/mapreduce_clusters.md]: suggest to named as `description`

File: docs/data-sources/mapreduce_clusters.md [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+  The cluster scale-in progress includes:
+    + **Preparing for scale-in**: Cluster scale-in is being prepared.
+    + **Decommissioning instance**: The instance is being decommissioned.
+    + **Deleting VMs**: The VMs are being deleted.
+    + **Deleting nodes from the cluster**: The nodes are being deleted from the cluster.
+    + **Scale-in failed**: Failed to scale in the cluster.
+
+  If the cluster installation, scale-out, or scale-in fails, stageDesc will display the failure cause.
+
+* `is_mrs_manager_finish` - Whether MRS Manager installation is finished during cluster creation.
```
Comment (ShiChangkuo) [inline docs/data-sources/mapreduce_clusters.md]: it this attribute useful?

File: huaweicloud/services/mrs/data_source_huaweicloud_mapreduce_clusters.go [hunk 3]
```diff
@@ -0,0 +1,681 @@
...
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: `Whether the subscription type is yearly or monthly.`,
+			},
+			"scale": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Status of node changes.`,
+			},
+			"eip_id": {
```
Comment (ShiChangkuo) [inline huaweicloud/services/mrs/data_source_huaweicloud_mapreduce_clusters.go:325]: what's the difference between `eip_id` and `external_ip`?

File: huaweicloud/services/mrs/data_source_huaweicloud_mapreduce_clusters.go [hunk 4]
```diff
@@ -0,0 +1,681 @@
...
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Specification ID of a Core node.`,
+			},
+			"availability_zone": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The AZ.`,
+			},
+			"instance_id": {
```
Comment (ShiChangkuo) [inline huaweicloud/services/mrs/data_source_huaweicloud_mapreduce_clusters.go]: what's the difference between `instance_id` and `id`?

File: docs/data-sources/mapreduce_clusters.md (hunk 1 above)
Comment (hmbig2) [reply docs/data-sources/mapreduce_clusters.md]: ok

File: huaweicloud/services/mrs/data_source_huaweicloud_mapreduce_clusters.go (hunk 4 above)
Comment (hmbig2) [reply huaweicloud/services/mrs/data_source_huaweicloud_mapreduce_clusters.go]: the values are the same, delete the instance_id

File: docs/data-sources/mapreduce_clusters.md [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+    + **Deleting VMs**: The VMs are being deleted.
+    + **Deleting nodes from the cluster**: The nodes are being deleted from the cluster.
+    + **Scale-in failed**: Failed to scale in the cluster.
+
+  If the cluster installation, scale-out, or scale-in fails, stageDesc will display the failure cause.
+
+* `is_mrs_manager_finish` - Whether MRS Manager installation is finished during cluster creation.  
+  The following options are supported:
+    + **true**: MRS Manager installation is finished.
+    + **false**: MRS Manager installation is not finished.
```
Comment (hmbig2) [inline docs/data-sources/mapreduce_clusters.md]: it's finished before the cluster is OK, so can delete

File: huaweicloud/services/mrs/data_source_huaweicloud_mapreduce_clusters.go (hunk 3 above)
Comment (hmbig2) [reply huaweicloud/services/mrs/data_source_huaweicloud_mapreduce_clusters.go:325]: external_ip is exist always

File: docs/data-sources/mapreduce_clusters.md (hunk 2 above)
Comment (hmbig2) [reply docs/data-sources/mapreduce_clusters.md]: yes, it's useful

**Pull request merged**